app/api/v2/views/validator.py

```diff
@@ -0,0 +1,29 @@
+"""Docstring for our validator"""
+
+
+class CheckRequired:
+    """This class check the required fields are filled"""
+    def __init__(self, payload):
+        self.payload = payload
+
+    def check_payload_signup(self):
+        """To check if payload is not empty and all values are provided"""
+        if not self.payload:
+            return False
+        if all(key in self.payload for key in ['email', 'password', 'role', 'name']):
+                values = CheckRequired(self.payload)
+                values_return = values.check_data_payload()
+                if values_return:
+                    return self.payload
+
+    def check_data_payload(self):
+        """To check all are not empty"""
+        if not any(value == "" for value in self.payload.values()):
+            return self.payload
```

Comment:
The methods `check_payload_signup` and `check_data_payload ` do almost the same thing - checking that no part of the payload is empty (that all are provided). Combine the two into one. You don' have to use both `all` and `.any`. Use one that works best.

Reply:
The check_payload_signup check for the keys if they exist but the check_data_payload checks for empty values on the keys